src/workers/search/project-search.js

```diff
@@ -4,32 +4,65 @@
 
 // Maybe reuse file search's functions?
 
+import getMatches from "./get-matches";
+
 export function findSourceMatches(source, queryText) {
-  const { id, loadedState, text } = source;
+  const { loadedState, text } = source;
   if (loadedState != "loaded" || !text || queryText == "") {
     return [];
   }
 
+  const modifiers = {
+    caseSensitive: false,
+    regexMatch: false,
+    wholeWord: false
+  };
+
   const lines = text.split("\n");
-  let result = undefined;
-  const query = new RegExp(queryText, "g");
-
-  const matches = lines
-    .map((_text, line) => {
-      const indices = [];
-
-      while ((result = query.exec(_text))) {
-        indices.push({
-          sourceId: id,
-          line: line + 1,
-          column: result.index,
-          match: result[0],
-          value: _text
-        });
-      }
-      return indices;
-    })
-    .filter(_matches => _matches.length > 0);
-
-  return [].concat(...matches);
+
+  return getMatches(queryText, text, modifiers).map(({ line, ch }) => {
+    const { value, c } = truncateLine(lines[line], ch);
+    return {
+      line: line + 1,
+      column: ch,
+      tc: c,
```

Comment:
what does `tc` stand for? Could we have a better name (it maps to the returned column from `truncateLine`, so maybe it could simply replace `column` ?)

Reply:
So there are two lines here, one is the original line(huge one). and column refers to the column in that line, since we need it to navigate on cliick.

And the is the new cropped line, and `tc` represents the column in the cropped string, since we need to highlight the match found.
I am so bad at naming things, I am sure you can find me a nice name :P

Reply:
so, looks like `tc` could be renamed to `matchIndex` or something similar?